### ISSUE TITLE
Nested ternaries are rewritten to mean something else than the original code

### ISSUE THREAD

**Issue by anddoutoi** [standard/standard#1262]
This is due to not taking into account [Operator Precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence) when rewriting the code.

Run the following code in the console:
```js
const foos = ['fubar']
const foo = undefined
const defaultFoos = ['snafu']
const fubar = foos ? foos : foo ? [foo] : defaultFoos
console.log(fubar)
```
Output is:
> ["fubar"]

Go to https://standardjs.com/demo.html, paste the code (or use this [demo github gist link](https://standardjs.com/demo.html?gist=7b95832005eb2fa46c10c0726b849e91)) and click _Correct errors using --fix_ .

The code is rewritten to:
```js
const foos = ['fubar']
const foo = undefined
const defaultFoos = ['snafu']
const fubar = foos || foo ? [foo] : defaultFoos
console.log(fubar)
```
Running this in a console outputs:
> [undefined]

Standard has changed the actual meaning of the code.

**Comment by LinusU** [comment]
This is not good, would you mind filing a bug upstream with eslint, and then linking that issue here? Thanks

**Comment by anddoutoi** [comment]
Maybe related:
- https://github.com/standard/standard/issues/866
- https://github.com/eslint/eslint/issues/8507

**Comment by LinusU** [comment]
Ah, that might be it. There is an open PR here to fix it: #1267

ping @feross

**Comment by LinusU** [comment]
This is fixed upstream and should be fixed in Standard 13, install with `npm install standard@next`
